Sof 121 python to influx db

prcostello opened the pull request:
main.py will communicate with an InfluxDB running in a separate shell simulating a local database. Please see the README.md file on running the tests.

prcostello committed:
Adding Client and README

base code to start

prcostello committed:
updated testing

messing around with InfluxDB

prcostello committed:
Updated Testing v2

interactive testing

prcostello committed:
Connected to DB, Instructions Updated

See README.md for instructions on testing.

spramij committed:
Update README.md

Formatting

andrewnash commented on Examples/PythonToInflux/client.py:
```diff
@@ -0,0 +1,70 @@
+    client = InfluxDBClient(host, port, user, password, dbname)
+
+    print("Create database: " + dbname)
+    client.create_database(dbname)
```
Maybe it's not that big of a deal because it's local hosted but it's good practice to incorporate error handling when connecting/ instantiating databases (i.e. putting line 33 in a try/ except statement). You can see they also do this in the examples documentation [here](https://github.com/influxdata/influxdb-python/blob/master/examples/tutorial_server_data.py)

andrewnash commented on Examples/PythonToInflux/client.py:
```diff
@@ -0,0 +1,70 @@
+            "measurement": "cpu_load_short",
+            "tags": {
+                "host": "server01",
+                "region": "us-west"
```
Do you actually have to provide a region when local hosting?

andrewnash commented on Examples/PythonToInflux/client.py:
```diff
@@ -0,0 +1,70 @@
+    client.create_database(dbname)
+
+    print("Create a retention policy")
+    client.create_retention_policy('awesome_policy', '3d', 3, default=True)
```
Should probably suggest a real policy? See [documentation](https://docs.influxdata.com/influxdb/v1.7/query_language/database_management/#retention-policy-management)

andrewnash commented on Examples/PythonToInflux/client.py:
```diff
@@ -0,0 +1,70 @@
+    client.write_points(json_body)
+
+    print("Querying data: " + query)
+    result = client.query(query)
```
Might be useful to add a couple sample queries, even if they are just comments i.e.
`query = "SELECT MEAN(value) FROM {} WHERE \
            time > now() - 10d GROUP BY time(500m)".format(metric)`

prcostello committed:
updated, error existing

when querying data error:

influxdb.exceptions.InfluxDBClientError: 400: {"error":"error parsing query: found -, expected ; at line 1, char 24"}

been trying to find my error but not having any luck with it, all that is remaining from comments on pull request

prcostello committed:
updated, error existing 2.0

forgot to stage a change

spramij reviewed (changes requested):
Not a big deal at all, but instead of running `influxd` in a terminal, might want to have the influx service started, using either
- `brew services start influxdb` for Mac
- `sudo systemctl start influxdb` for Linux

Or even better, the services enabled, to start at boot time.

prcostello committed:
final update

query fixed, other housekeeping done.
